src/main/java/org/springframework/retry/annotation/RecoverAnnotationRecoveryHandler.java

```diff
@@ -81,25 +83,42 @@ public T recover(Object[] args, Throwable cause) {
 	}
 
 	private Method findClosestMatch(Object[] args, Class<? extends Throwable> cause) {
-		int min = Integer.MAX_VALUE;
 		Method result = null;
-		for (Method method : this.methods.keySet()) {
-			SimpleMetadata meta = this.methods.get(method);
-			Class<? extends Throwable> type = meta.getType();
-			if (type == null) {
-				type = Throwable.class;
+
+		boolean shouldUseRecoverMethodName = !this.recoverMethodName.equals("");
+		if (shouldUseRecoverMethodName) {
+			for (Method method : this.methods.keySet()) {
```

Comment:
The map iteration must be done over `entrySet`.
This way we won't have an overhead requesting each entry value with the next `get`.

Reply:
Agree. Fixed.